lib/utils.pm

```diff
@@ -1164,6 +1164,7 @@ sub ensure_serialdev_permissions {
     }
     else {
         assert_script_run "chown $testapi::username /dev/$testapi::serialdev && gpasswd -a $testapi::username \$(stat -c %G /dev/$testapi::serialdev)";
+        assert_script_run "gpasswd -a $testapi::username dialout";
```

Comment:
We already do that in the line above. On a recent system `stat -c %G /dev/$testapi::serialdev` returns "dialout". This is for backward-compatibility with older products which had AFAIR *tty*. We should not need this line. Which tests failed for you that you think you need this fix?

Reply:
But after installation and reboot, the /dev/ttyS0 would change group to "dialout". Please reference [poo#38531](https://progress.opensuse.org/issues/38531). This caused by "systemctl --no-pager mask serial-gettty@ttyS0 and systemctl --no-pager stop serial-gettty@ttyS0" in "consoletest_setup".

Reply:
It will cause users and /dev/ttyS0 belongs to different groups. So the user have no permission to write to /dev/ttyS0.

Reply:
All failed test you have mentioned in [poo#38531](https://progress.opensuse.org/issues/38531) and [poo#38480](https://progress.opensuse.org/issues/38480) all because this reason.

Reply:
Hm, in https://openqa.opensuse.org/tests/708129#step/consoletest_setup/6 the group is tty.

@yuanren10 so you are saying that masking the serial getty causes the group of the device to change? That would be interesting.